libs/environment/src/hooks/use-nodes.spec.tsx

```diff
@@ -0,0 +1,402 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import { ApolloClient } from '@apollo/client';
+import createClient from '../utils/apollo-client';
+import { useNodes } from './use-nodes';
+import { Networks } from '../types';
+import createMockClient, {
+  getMockStatisticsResult,
+} from './mocks/apollo-client';
+
+jest.mock('../utils/apollo-client');
+
+const MOCK_ENV = Networks.DEVNET;
+const MOCK_DURATION = 1073;
+
+const initialState = {
+  url: '',
+  verified: false,
+  initialized: false,
+  responseTime: {
+    isLoading: false,
+    hasError: false,
+    value: undefined,
+  },
+  block: {
+    isLoading: false,
+    hasError: false,
+    value: undefined,
+  },
+  ssl: {
+    isLoading: false,
+    hasError: false,
+    value: undefined,
+  },
+  chain: {
+    isLoading: false,
+    hasError: false,
+    value: undefined,
+  },
+};
+
+window.performance.getEntriesByName = jest
+  .fn()
+  .mockImplementation((url: string) => [
+    {
+      entryType: 'resource',
+      name: url,
+      startTime: 0,
+      toJSON: () => ({}),
+      duration: MOCK_DURATION,
+    },
+  ]);
+
+beforeEach(() => {
+  // @ts-ignore allow adding a mock return value to mocked module
+  createClient.mockImplementation(() => createMockClient());
```

Comment:
question: why is this required? is it not possible for us to use MockedProvider like in other tests?